python/cudf/cudf/dataframe/series.py

```diff
@@ -1194,6 +1191,18 @@ def notna(self):
         mask = cudautils.notna_mask(self.data, self.nullmask.to_gpu_array())
         return Series(mask, name=self.name, index=self.index)
 
+    def nans_to_nulls(self):
+        """
+        Convert nans (if any) to nulls
+        """
+        from cudf.utils.cudautils import mask_from_devary
+
+        if self.dtype.kind == "f":
+            sr = self.fillna(np.nan)
+            newmask = mask_from_devary(sr._column.data.mem)
```

Comment:
This will set the currently null bits in the mask to valid, you need to merge the new mask with an existing mask.

Reply:
Not sure I follow: the approach is to first replace any existing nulls with NaNs and then generate a new bitmask which contains nulls everywhere there are NaNs in the data. Am I missing something?

Reply:
May be wrong, but I think @kkraus14 is suggesting that instead of nulls > nan, nan => nulls, we could do something like `original.replace(mask=original.nullmask.mem & compact_mask_bytes(original == nan)` ?

Reply:
@cwharris - IIRC @kkraus14 and I discussed offline and agreed that the approach above works. Apologies for not updating here